addons/binding/org.openhab.binding.mihome/ESH-INF/thing/channel.xml

```diff
@@ -79,6 +79,13 @@
 	<channel-type id="joinPermission">
 		<item-type>Switch</item-type>
 		<label>Enable Device Inclusion</label>
+		<description>Enable pairing for 30 seconds. You then have to press the pairing button on the device.</description>
+	</channel-type>
+
+	<channel-type id="removeDevice">
```

Comment:
@kaikreuzer this looks strange to me, what do you think?

Reply:
@martinvw please explain strange 😉

Reply:
Fair question, I would not expect to do removing of devices from a gateway through rules, but more through the UI or something, but maybe I'm wrong :-)

Reply:
I see. Any recommendations? I imagine having sort of a button in the thing config in PaperUI to trigger device inclusion/removal. Actually I would like see such a functionality implemented for bridges in ESH. So binding developers only have to implement the methods.

Reply:
You maybe check the z-wave binding and or chat with @cdjackson :-) Z-wave and Zigbee have similar problems I would guess

Reply:
I did check the zigbee binding. It basically tries to remove the device from the network as soon as it is removed in PaperUI. 
I didn't want to mess with the gateway so much as people would also have the MiHome App in parallel. So removing the device automatically with removing the thing I don't think is needed.

Reply:
@kaikreuzer any preference / opinion here?

Reply:
I probably do not grasp what exactly is the use case here. Is it that the user wants to remove a device (=Thing) from openHAB and assumes that at the same time it is unpaired from the MiHome gateway? If so, the standard mechanism is that the user simply deletes the Thing through the Paper UI, nothing more than that. The binding should then implement the `ThingHandler.handleRemoval()` method and remove the device from the gateway.

Reply:
Thanks for your opinions on this. You are right, there shouldn't be any confusing workarounds. Just add or remove devices/things.
I have edited the commit and removed the two channels (join permission & remove device) from the gateway thing.
The permission to join the network is now triggered during the bindings thing discovery. As soon as the user removes a thing, the gateway gets the command to remove it from the network as well (in the `handleRemoval` method 😄 
There is no check if the gateway really removed it though, so the thing status is changed to `REMOVED` immediately.

I also did a small fix for the fingerprint lock, I hope you guys let me sneak that in...
Happy new year and thank you very much for making the clocks tick ❤️